src/app/views/sidebar/sample-queries/sample-query-utils.ts

```diff
@@ -48,3 +48,23 @@ export const trackDocumentLinkClickedEvent = async (item: ISampleQuery): Promise
   // Check if link throws error
   validateExternalLink(item.docLink || '', componentNames.DOCUMENTATION_LINK, item.id);
 }
+
+interface ShouldRunQueryArgs {
+  method: string;
+  url: string;
+  authenticated: boolean;
+}
+
+const shouldRunQuery = ({ method, url, authenticated }: ShouldRunQueryArgs): boolean => {
```

Comment:
This method should return true if we should run a query. Making it return false will be confusing for future maintenance.
In QueryInput.tsx, we should have something like this `const showError = !shouldRunQuery({
    method: sampleQuery.selectedVerb, authenticated,
    url: sampleQuery.sampleUrl
  });`
This means we should not show the error when shouldRunQuery returns true. 

Also, in SampleQueries.tsx:
` if (!shouldRunQuery({ method: query.method, authenticated: tokenPresent, url: query.requestUrl })) {
        selectionDisabled = true;
      }`
This means the selection is disabled only when we should not run the query.
What do you think?

Reply:
makes sense...

Reply:
This makes sense. Thanks for catching it @Onokaev and @ElinorW